Code/Framework/AzToolsFramework/Tests/TransformComponent.cpp

```diff
@@ -1063,4 +1067,53 @@ R"DELIMITER(<ObjectStream version="1">
             }
         }
     }
+
+    // Fixture provides TransformComponent that is static (or not static) on an entity that has been activated.
+    class TransformComponentActivationTest
+        : public PrefabTestFixture
+        , public TransformNotificationBus::Handler
+    {
+    protected:
+        void SetUpEditorFixtureImpl() override
+        {
+            PrefabTestFixture::SetUpEditorFixtureImpl();
+
+            CreateRootPrefab();
+        }
+
+        void TearDownEditorFixtureImpl() override
+        {
+            BusDisconnect();
+            
+            PrefabTestFixture::TearDownEditorFixtureImpl();
+        }
+
+        //! TransformNotificationBus::Handler implementation ...
+        void OnTransformChanged(const Transform& /*local*/, const Transform& /*world*/) override
+        {
+            m_transformUpdated = true;
+        }
+
+        void MoveEntity(AZ::EntityId entityId)
+        {
+            AzToolsFramework::ScopedUndoBatch undoBatch("Move Entity");
+            TransformBus::Event(entityId, &TransformInterface::SetWorldTranslation, Vector3(1.f, 0.f, 0.f));
+        }
+        
+        bool m_transformUpdated = false;
+    };
+    
+    TEST_F(TransformComponentActivationTest, OnTransformChanged_OnEntityActivated)
```

Comment:
suggestion: Awesome to see the tests but it would be great to have a more descriptive test name to describe what behavior this test is actually validating. Perhaps something like `TransformChangedEventIsSentWhenEntityIsActivatedViaUndoRedo` (we should also aim to avoid underscores in test names as those are reserved characters by Google Test)